### PR TITLE
reimplement const pool and reenable const jumps and jump dispatch opt.

### PR DESCRIPTION
**Pull request by gcolvin**
With a maximum size on code the constant pool can now be a vector of all constants, quickly constructed and not exploitable at runtime - unlike the previous small table and hash table.  Also turned back on the other optimizations, and added new opcodes to the dispatch table and debugged it.

### CODE REVIEW

File: libevm/VM.cpp [hunk 1]
```diff
@@ -15,8 +15,6 @@
 	along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
 */
 /** @file VM.cpp
```
Comment (chfast) [inline libevm/VM.cpp]: Remove the full comment section.

File: libevm/VM.h [hunk 2]
```diff
@@ -15,8 +15,6 @@
 	along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
 */
 /** @file VM.h
```
Comment (chfast) [inline libevm/VM.h]: Remove full comment section.

File: test/unittests/performance/tests.mk [hunk 3]
```diff
@@ -26,9 +26,9 @@ endif
 	$(TEST_ETHVM) $@
 	$(TEST_EVM) $@ run
 
-% : %.c
-	$(CCC) $* $^
-	$(STATS) ./$*
+mul64c: mul64c.c
```
Comment (chfast) [inline test/unittests/performance/tests.mk]: Can you add some description in the beginning of this file how to use it?

Comment (gcolvin) [inline test/unittests/performance/tests.mk]: OK, will sneak into some semi-relevant PR.

File: libevm/VM.cpp (hunk 1 above)
Comment (gcolvin) [inline libevm/VM.cpp]: That is, the entire copyright section?  Or just section starting with `/**`  ?

**Pull request merged**